docs/testplan/MACsec-test-plan.md

```diff
@@ -0,0 +1,445 @@
+```
+
+## Common Configuration
+
+### Dynamic Key(MKA)
+
+***MACsec profile table***
+
+| Field                 |                                                              Value                                                              |
+| --------------------- | :-----------------------------------------------------------------------------------------------------------------------------: |
+| priority              |                                                  DUT(*64*) VM0(*63*) VM1(*65*)                                                  |
+| cipher suite          |                                                   *GCM-AES-128*/*GCM-AES-256*                                                   |
+| CKN                   |                               *6162636465666768696A6B6C6D6E6F707172737475767778797A303132333435*                                |
+| CAK                   | GCM-AES-128(*0123456789ABCDEF0123456789ABCDEF*)/GCM-AES-256(*0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF*) |
+| policy                |                                                           *security*                                                            |
+| enable_replay_protect |                                                         *true*/*false*                                                          |
+| replay_window         |                                    enable_replay_protect(*0*)/disable_replay_protect(*100*)                                     |
+| send_sci              |                                                         *true*/*false*                                                          |
+| rekey_period          |                                                            *0*/*60*                                                             |
+
+***Port table***
+| Field               |                Value                |
+| ------------------- | :---------------------------------: |
+| pfc_encryption_mode | *bypass*/*encrypt*/*strict_encrypt* |
+
+**NOTE: The macsec functionality testcases will cover the various options for each of the above supported fields**
+
+## Test cases
+
+### Testcase : Macsec Functionality
+
+This testcase covers the macsec/MKA protocol functionality
+
+#### Check Control plane
+
+- Enable macsec on the interface on the DUT and remote VM.
+- Check the process, `wpa_supplicant`, for the target port is running in the devices.
+- Check APP DB, Check the following fields in MACsec port table are consistent with configuration
+
+    | Config DB Field | Config DB Value |  App DB Field  | APP DB Value |
+    | :-------------: | :-------------: | :------------: | :----------: |
+    |                 |                 |     enable     |     true     |
+    |  cipher_suite   |   GCM-AES-128   |  cipher_suite  | GCM-AES-128  |
+    |  cipher_suite   |   GCM-AES-256   |  cipher_suite  | GCM-AES-256  |
```

Comment:
Need an update for XPN cipher suite in this table.

Reply:
Thanks for your suggestion, added it.